opensrp-stock/src/main/java/org/smartregister/stock/repository/StockRepository.java

```diff
@@ -231,9 +236,9 @@ public List<Stock> findUnSyncedWithLimit(int limit) {
         return stocks;
     }
 
-    public List<Stock> findUniqueStock(String stock_type_id, String transaction_type, String providerid, String value, String date_created, String to_from) {
+    public List<Stock> findUniqueStock(String identifier, String stock_type_id, String transaction_type, String providerid, String value, String to_from) {
         List<Stock> stocks = new ArrayList<>();
-        try (Cursor cursor = getReadableDatabase().query(STOCK_TABLE_NAME, STOCK_TABLE_COLUMNS, STOCK_TYPE_ID + " = ? AND " + TRANSACTION_TYPE + " = ? AND " + PROVIDER_ID + " = ? AND " + VALUE + " = ? AND " + DATE_CREATED + " = ? AND " + TO_FROM + " = ?", new String[]{stock_type_id, transaction_type, providerid, value, date_created, to_from}, null, null, null, null)) {
+        try (Cursor cursor = getReadableDatabase().query(STOCK_TABLE_NAME, STOCK_TABLE_COLUMNS, IDENTIFIER + " = ? AND " + STOCK_TYPE_ID + " = ? AND " + TRANSACTION_TYPE + " = ? AND " + PROVIDER_ID + " = ? AND " + VALUE + " = ? AND " + TO_FROM + " = ?", new String[]{identifier, stock_type_id, transaction_type, providerid, value, to_from}, null, null, null, null)) {
```

Comment:
Given that we added a unique identifier for each of the entries, should we filter by all others?